### DISCUSSION TITLE
RequestError and AuthenticationException when trying to redeploy elastalert

### DISCUSSION THREAD

**Discussion by shireeshac01** [jertel/elastalert2#980] [Q&A] [answered]
After deleting all the existing elastalert indices, we tried redploying elastalert from scratch. But in logs we keep seeing the following Exceptions inspite of providing correct credentials. We verified curl request to elasticsearch with same credentials and it is successful

`ERROR:elastalert:Error querying for last run: RequestError(400, 'search_phase_execution_exception', 'No mapping found for [@timestamp] in order to sort on')
WARNING:elasticsearch:POST https://<elasticsearch>:443/elastalert_status/_search?_source_includes=endtime%2Crule_name&size=1 [status:401 request:0.034s]
ERROR:elastalert:Error querying for last run: AuthenticationException(401, 'security_exception', 'unable to authenticate user [admin] for REST request [/elastalert_status/_search?_source_includes=endtime%2Crule_name&size=1]')
WARNING:elasticsearch:POST https://<elasticsearch>:443/ops.os4.kube.ech-rp*/_search?ignore_unavailable=true&size=0 [status:401 request:0.126s]`

Could anyone suggest how can we resolve this issue

**Comment by jertel** [comment, marked as answer]
I suggest checking the Elasticsearch logs to see if they offer additional information on the auth error. Otherwise, if you're using ElastAlert 2 via Docker, exec into the running container and verify the credentials in the mapped config file are exactly as expected. It may also be useful to enclose the password in double quotes if it contains non-alphanumeric characters.

Also, for production systems I suggest creating a dedicated elastalert2 user in Elasticsearch, instead of using the admin account. That way you can lock it down to only having write permissions to the `elastalert_*` indices, and only provide read access to the subset of indices it should be reading.

**Reply by praneethkaturi** [reply]
Elastalert2 is running as a kuberentes pod; we have exec'd into the pod and verified the credentials are correct. We performed the curl request to the elasticsearch server using the environment variables in the pod. 
We've created a new "elastalert2" user as well, and the same error seems to persist.

**Reply by jertel** [reply]
How about the ES log? Anything in there about these failed requests?

You mentioned environment variables in the pod. How are you specifying the username/password? Typically those credentials are specified in the values.yaml override, or in a deployed secret.
